spec/controllers/communicarts_controller_spec.rb

```diff
@@ -265,21 +263,23 @@
     context 'valid params' do
       before do
         expect(ApiToken).to receive(:find_by).with(access_token: "5a4b3c2d1ee1d2c3b4a5").and_return(token)
+        approver
+        expect_any_instance_of(Approval).to receive(:update_attributes)
       end
 
       it 'will be successful' do
-        approver
-        expect_any_instance_of(Approval).to receive(:update_attributes)
         put 'approval_response', @json_approval_params_with_token
-        expect(response.status).to eq 200
+        expect(response).to redirect_to(cart_path(cart))
       end
 
       it 'successfully validates the user_id and cart_id with the token' do
-        approver
-        expect_any_instance_of(Approval).to receive(:update_attributes)
         expect { put 'approval_response', @json_approval_params_with_token }.not_to raise_error
       end
 
+      it "signs the user in" do
+        put 'approval_response', @json_approval_params_with_token
```

Comment:
Ha, no, `put` as in "HTTP PUT".

http://api.rubyonrails.org/classes/ActionController/TestCase/Behavior.html#method-i-put